canonical/loki-k8s-operator
# Support juju proxy settings

## Issue
Fixes https://github.com/canonical/loki-k8s-operator/issues/326

## Solution
I declared a ProxyHandler for the urllib request that is made to download the promtail binary.

## Testing Instructions
I used the discourse-k8s charm to test the change.  
1. Add the cilium addon to microk8s
2. Deploy squid on microk8s like proposed [here](https://hub.docker.com/r/ubuntu/squid)
3. Create a model
4. apply a network-policy to that model only allowing local networking (like [this](https://gist.github.com/nrobinaubertin/f583409700bced7edf03827d3fd48134)) 
5. Apply the correct juju-http{s,}-proxy settings to the model
6. relate all the charms to discourse-k8s, it should be operational in the end.


## Release Notes
Add support for JUJU_CHARM_{HTTP,HTTPS}_PROXY variables in the loki_push_api lib.
